src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,312 @@
+              .map((res) => {
+                return {
+                  // res has been filtered and is always a response event.
+                  // TS doesn't guess.
+                  manifest: (res as ILoaderResponse<Document>).value.responseData,
+                  url: content.url,
+                  startTime: content.startTime,
+                  endTime: content.endTime,
+                  transport: content.transport,
+                };
+              });
+          });
+
+          return Observable
+            .combineLatest(manifestsInfos$)
+            .map((manifestsInfos) => {
+              return {
+                type: "response" as "response",
+                value: {
+                  responseData: {
+                    manifests: manifestsInfos,
+                  },
+                },
+              };
+            });
+        });
+      },
+
+      parser(
+        { response, url } : IManifestParserArguments<IMetaManifestInfo>
+      ) : IManifestParserObservable {
+
+        const documents = response.responseData;
+        const manifestsInfos =  documents.manifests;
+
+        const parsedManifestsInfo = manifestsInfos.map((manifestInfos) => {
+          const transport = transports[manifestInfos.transport];
+          const parser = transport.manifest.parser;
+          const fakeResponse = {
+            responseData: manifestInfos.manifest,
+          };
+          return parser({ response: fakeResponse, url: manifestInfos.url })
+          .map((mpd) => {
+            const type = mpd.manifest.type;
+            if (type !== "static") {
+              throw new Error("Content from metaplaylist is not static.");
+            }
+            return {
+              manifest: mpd.manifest,
+              transport: manifestInfos.transport,
+              url: manifestInfos.url,
+              startTime: manifestInfos.startTime,
+              endTime: manifestInfos.endTime,
+            };
+          });
+        });
+
+        return Observable.combineLatest(parsedManifestsInfo).map((contents) => {
+            const manifest = generateManifest(contents, url);
+            return {
+              manifest,
+              url,
+            };
+        });
+      },
+    };
+
+    const segmentPipeline = {
+      loader({
+        adaptation,
+        init,
+        manifest,
+        period,
+        representation,
+        segment,
+      } : ISegmentLoaderArguments) : ILoaderObservable<Uint8Array|ArrayBuffer> {
+        if (!segment.privateInfos) {
+          throw new Error("Segments from metaplaylist must have private infos.");
+        }
+        const transportType = getTypeFromPrivateInfos(segment.privateInfos);
+        const transport = transports[transportType];
+        const segmentLoader = transport.video.loader;
+        return segmentLoader({
+          adaptation,
+          init,
+          manifest,
+          period,
+          representation,
+          segment,
+        });
```

Comment:
Isn't it dangerous to not mutate segment's data to what it is supposed to be here?

As in (pushed values == with tfdt modified, initial values == manifest values):
  - Things that goes through DASH/SMOOTH transports are always using the initial values without even being aware of the pushed ones
  - Translations between pushed values and initial values happen only when it goes through the wrapper (representationIndex definition + various loaders and parsers)
  - the rest of the code are only using the pushed values without even being aware of the initial ones

Note: We might need to clone segments here to avoid some ugly side-effects. We might want to make sure that cloning do not break anything in the RepresentationBuffer -> pipeline -> net (etc.) workflow.

I don't think it can go wrong under that strategy